lib/internal/fs/glob.js

```diff
@@ -0,0 +1,371 @@
+'use strict';
+const { lstatSync, readdirSync } = require('fs');
+const { join, resolve } = require('path');
+
+const {
+  kEmptyObject,
+} = require('internal/util');
+const {
+  validateFunction,
+  validateObject,
+} = require('internal/validators');
+
+const {
+  ArrayFrom,
+  ArrayPrototypeAt,
+  ArrayPrototypeMap,
+  ArrayPrototypeFlatMap,
+  ArrayPrototypePop,
+  ArrayPrototypePush,
+  ArrayPrototypeSome,
+  SafeMap,
+  SafeSet,
+  StringPrototypeEndsWith,
+} = primordials;
+
+let minimatch;
+function lazyMinimatch() {
+  minimatch ??= require('internal/deps/minimatch/index');
+  return minimatch;
+}
+
+const isWindows = process.platform === 'win32';
+const isOSX = process.platform === 'darwin';
+
+class Cache {
+  #cache = new SafeMap();
+  #statsCache = new SafeMap();
+  #readdirCache = new SafeMap();
+
+  statSync(path) {
+    if (this.#statsCache.has(path)) {
+      return this.#statsCache.get(path);
```

Comment:
Instead of a `has()` followed by a `get()`, can you just do a `get()` and check against `undefined`.